code/modules/vehicles/mecha/equipment/weapons/greyscale_weapons.dm

```diff
@@ -421,3 +421,88 @@
+#define LASER_DASH_RANGE_ENHANCED 5
+
+/obj/item/mecha_parts/mecha_equipment/laser_sword
+	name = "\improper Moonlight particle cutter"
+	icon = 'icons/mecha/mecha_equipment_64x32.dmi'
+	desc = "A specialized mech laser blade made out of compressed energy with unimaginable power. Its compact size allows fast, short-ranged attacks. When activated, overloads the leg actuators to dash forward, before cutting with a superheated plasma beam. Melee core increases area cut and distance dashed. Heavy, but it is the top-of-the-line melee weapon of TGMC's fine line of mecha close-range offensive capability."
+	icon_state = "moonlight"
+	mech_flags = EXOSUIT_MODULE_GREYSCALE
+	obj_integrity = 400
+	slowdown = 0
+	harmful = TRUE
+	equip_cooldown = 1 SECONDS
+	energy_drain = 0 // energy drain due to dashing is good enough
+	range = MECHA_MELEE|MECHA_RANGED
+	force = 150
+	/// holder var for the mob that is attacking right now
+	var/mob/cutter
+
+/obj/item/mecha_parts/mecha_equipment/laser_sword/action_checks(atom/target, ignore_cooldown)
+	. = ..()
+	if(!.)
+		return
+	if(chassis.zoom_mode)
+		to_chat(chassis.occupants, "[icon2html(src, chassis.occupants)][span_warning("Unable to dash while in zoom mode!")]")
+		return FALSE
+	if(cutter)
+		to_chat(chassis.occupants, "[icon2html(src, chassis.occupants)][span_warning("Already in use!")]")
+		return FALSE
+
+/obj/item/mecha_parts/mecha_equipment/laser_sword/action(mob/source, atom/target, list/modifiers)
+	if(!action_checks(target))
+		return
+	//melee swipe, no need to dash
+	if(chassis.Adjacent(target))
+		execute_melee(source, modifiers)
+		return ..()
+
+	//try dash to target
+	var/laser_dash_range = LASER_DASH_RANGE_ENHANCED
+
+	chassis.add_filter("dash_blur", 1, radial_blur_filter(0.3))
+	icon_state += "_on"
+	chassis.update_icon()
+	new /obj/effect/temp_visual/xenomorph/afterimage(chassis.loc, chassis)
+	RegisterSignal(chassis, COMSIG_MOVABLE_POST_THROW, .proc/end_dash)
+	cutter = source
+	chassis.flags_atom |= DIRLOCK
+	RegisterSignal(chassis, COMSIG_MOVABLE_MOVED, .proc/drop_afterimage)
+	chassis.throw_at(target, laser_dash_range, 1, flying = TRUE)
+	return ..()
+
+///signal handler, drops afterimage every move executed while dashing
+/obj/item/mecha_parts/mecha_equipment/laser_sword/proc/drop_afterimage(datum/source)
+	SIGNAL_HANDLER
+	new /obj/effect/temp_visual/xenomorph/afterimage(chassis.loc, chassis)
+
+///Ends dash and executes attack
+/obj/item/mecha_parts/mecha_equipment/laser_sword/proc/end_dash(datum/source)
+	SIGNAL_HANDLER
+	UnregisterSignal(source, list(COMSIG_MOVABLE_POST_THROW, COMSIG_MOVABLE_MOVED))
+	chassis.remove_filter("dash_blur")
+	icon_state = initial(icon_state)
+	chassis.update_icon()
+	execute_melee(cutter)
+	cutter = null
+	chassis.flags_atom &= ~DIRLOCK
+
+///executes a melee attack in the direction that the mech is facing
+/obj/item/mecha_parts/mecha_equipment/laser_sword/proc/execute_melee(mob/source, list/modifiers)
+	var/turf/target = get_step(chassis, chassis.dir)
```

Comment:
Does this mean you can't diag slash even though dash does work diagonally? And depending on the path of the throw, I think that means it's possible for the throw to hit something, but you don't actually swing at it.

Reply:
If you do almost directly up yes, but  making it diag slash would be even more hacky

Reply:
More generally that makes melee quite difficult, since xenos can just strafe and hit you on the diag but you can't hit them back.
Is that intentional?